deploy/s3-bucket-head.sh

```diff
@@ -1,14 +1,25 @@
-#!bin/bash
+#!/bin/bash
 
 set -e
 
-export TMP_S3_HOST=`echo $TMP_S3_ENDPOINT | awk -F/ '{print $3}'`
-export TMP_S3_S3CMD_FLAGS="--access_key=$TMP_S3_ACCESS_KEY_ID --secret_key=$TMP_S3_SECRET_ACCESS_KEY --host=$TMP_S3_HOST --no-ssl --host-bucket=$TMP_S3_BUCKET_NAME.$TMP_S3_HOST"
+TMP_S3_S3CMD_SSL_FLAGS=""
+if [ -z "$TMP_CA_CERT" ] && echo "$TMP_S3_ENDPOINT" | grep -q "http://"; then
+  TMP_S3_S3CMD_SSL_FLAGS="--no-ssl"
+else
+    echo "$TMP_CA_CERT" > /tmp/cabundle.crt
+    TMP_S3_S3CMD_SSL_FLAGS="--ca-certs=/tmp/cabundle.crt"
+fi
+
+TMP_S3_HOST=$(echo "$TMP_S3_ENDPOINT" | awk -F/ '{print $3}')
+export TMP_S3_HOST
+export TMP_S3_S3CMD_FLAGS="--access_key=$TMP_S3_ACCESS_KEY_ID --secret_key=$TMP_S3_SECRET_ACCESS_KEY --host=$TMP_S3_HOST --host-bucket=$TMP_S3_BUCKET_NAME.$TMP_S3_HOST $TMP_S3_S3CMD_SSL_FLAGS"
 
-if s3cmd $TMP_S3_S3CMD_FLAGS ls s3://$TMP_S3_BUCKET_NAME 2>&1 | grep -q 'NoSuchBucket'
+echo "Running with flags: --access_key=***** --secret_key=***** --host=$TMP_S3_HOST --host-bucket=$TMP_S3_BUCKET_NAME.$TMP_S3_HOST $TMP_S3_S3CMD_SSL_FLAGS"
+
+if timeout 5 s3cmd $TMP_S3_S3CMD_FLAGS ls s3://$TMP_S3_BUCKET_NAME
```

Comment:
Added the timeout here. If you type in the wrong IP address s3cmd will block and keep retrying.

Reply:
can we increase the timeout a bit? 5 seconds seem very low

Reply:
15 per our slack thread 👍